sp-sim/src/server.rs

```diff
@@ -56,9 +56,10 @@ impl UdpServer {
                     bail!("{multicast_addr} is not multicast!");
                 }
 
-                let interface_index =
-                    if_nametoindex(multicast_interface.as_str())
-                        .context("if_nametoindex for {multicast_interface}")?;
+                let interface_index = if_nametoindex(
+                    multicast_interface.as_str(),
+                )
+                .context(format!("if_nametoindex for {multicast_interface}"))?;
```

Comment:
Oof, I thought clippy warned about string literals with `{braced_items}` that weren't inside `format!`.

Since this allocates a string, I think this should be

```suggestion
                .with_context(|| format!("if_nametoindex for {multicast_interface}"))?;
```

Reply:
I also thought this! changed to with_context in a5b815bb178bf561d59e52e5f862c3ef70b4020b